lib/Controller/SignalingController.php

```diff
@@ -104,6 +109,54 @@ public function getSettings(): DataResponse {
 		return new DataResponse($this->config->getSettings($this->userId));
 	}
 
+	/**
+	 * Only available for logged in users because guests can not use the apps
+	 * right now.
+	 *
+	 * @param int $serverId
+	 * @return DataResponse
+	 */
+	public function getWelcomeMessage(int $serverId): DataResponse {
+		$signalingServers = $this->config->getSignalingServers();
+		if (empty($signalingServers) || !isset($signalingServers[$serverId])) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		$url = rtrim($signalingServers[$serverId]['server'], '/');
+
+		if (strpos($url, 'wss://') === 0) {
+			$url = 'https://' . substr($url, 6);
+		}
+
+		if (strpos($url, 'ws://') === 0) {
+			$url = 'http://' . substr($url, 5);
+		}
+
+		$client = $this->clientService->newClient();
+		try {
+			$response = $client->get($url . '/api/v1/welcome', [
+				'verify' => (bool) $signalingServers[$serverId]['verify'],
```

Comment:
I have checked the external signaling server code and I could not find where it is used (although it may be used by a library, I do not know). The only thing that I could find was the `skipverify` parameter, but as far as I know that comes from the settings and not the requests. @fancycode Is the `verify` parameter needed? Thanks for the clarification!

Reply:
this is used in the GuzzleClient to allow access custom SSL certs

Reply:
Yes, this is so connections to untrusted (e.g. self-signed) certificates work. Only used by the Nextcloud HTTP client.